net-im/gitter-bin/gitter-bin-4.1.0-r1.ebuild

```diff
@@ -0,0 +1,107 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+MULTILIB_COMPAT=( abi_x86_{32,64} )
+MY_PN="${PN/-bin/}"
+
+inherit desktop multilib-build pax-utils unpacker xdg-utils
+
+QA_PREBUILT="opt/gitter/pnacl/pnacl_public_x86_64_libcrt_platform_a
+	opt/gitter/pnacl/pnacl_public_x86_64_pnacl_llc_nexe
+	opt/gitter/pnacl/pnacl_public_x86_64_ld_nexe
+	opt/gitter/pnacl/pnacl_public_x86_64_pnacl_sz_nexe
+	opt/gitter/payload
+	opt/gitter/swiftshader/libEGL.so
+	opt/gitter/swiftshader/libGLESv2.so
+	opt/gitter/chromedriver
+	opt/gitter/lib/libnw.so
+	opt/gitter/lib/libnode.so
+	opt/gitter/lib/libffmpeg.so
+	opt/gitter/nacl_helper
+	opt/gitter/nwjc
+	opt/gitter/nacl_irt_x86_64.nexe
+	opt/gitter/Gitter"
+
+DESCRIPTION="Chat and network platform"
+HOMEPAGE="https://www.gitter.im"
+SRC_URI="
+	amd64? ( https://update.gitter.im/linux64/${MY_PN}_${PV}_amd64.deb )
+	x86? ( https://update.gitter.im/linux32/${MY_PN}_${PV}_i386.deb )"
+
+LICENSE="all-rights-reserved"
+SLOT="0"
+KEYWORDS="-* ~amd64 ~x86"
+IUSE=""
+RESTRICT="bindist mirror"
+
+RDEPEND="dev-libs/atk:0[${MULTILIB_USEDEP}]
+	dev-libs/expat:0[${MULTILIB_USEDEP}]
+	dev-libs/glib:2[${MULTILIB_USEDEP}]
+	dev-libs/nspr:0[${MULTILIB_USEDEP}]
+	dev-libs/nss:0[${MULTILIB_USEDEP}]
+	media-libs/alsa-lib:0[${MULTILIB_USEDEP}]
+	sys-apps/dbus:0[${MULTILIB_USEDEP}]
+	x11-libs/cairo:0[${MULTILIB_USEDEP}]
+	x11-libs/gdk-pixbuf:2[${MULTILIB_USEDEP}]
+	x11-libs/gtk+:3[${MULTILIB_USEDEP}]
+	x11-libs/libX11:0[${MULTILIB_USEDEP}]
+	x11-libs/libxcb:0/1.12[${MULTILIB_USEDEP}]
+	x11-libs/libXcomposite:0[${MULTILIB_USEDEP}]
+	x11-libs/libXcursor:0[${MULTILIB_USEDEP}]
+	x11-libs/libXdamage:0[${MULTILIB_USEDEP}]
+	x11-libs/libXext:0[${MULTILIB_USEDEP}]
+	x11-libs/libXfixes:0[${MULTILIB_USEDEP}]
+	x11-libs/libXi:0[${MULTILIB_USEDEP}]
+	x11-libs/libXrandr:0[${MULTILIB_USEDEP}]
+	x11-libs/libXrender:0[${MULTILIB_USEDEP}]
+	x11-libs/libXtst:0[${MULTILIB_USEDEP}]
+	x11-libs/pango:0[${MULTILIB_USEDEP}]"
+BDEPEND="sys-apps/fix-gnustack"
+
+S="${WORKDIR}"
+
+src_prepare() {
+	default
+
+	local arch
+	arch="$(usex amd64 "64" "32")"
+
+	# Modify desktop file to use common paths
+	sed -i \
+		-e '/Exec/s/=.*/=\/usr\/bin\/gitter/' \
+		-e '/Icon/s/=.*/=\/usr\/share\/pixmaps\/gitter.png/' \
+		opt/Gitter/linux"${arch}"/gitter.desktop || die "sed failed"
+}
+
+src_install() {
+	local arch
+	arch="$(usex amd64 "64" "32")"
+
+	fix-gnustack -f opt/Gitter/linux"${arch}"/nacl_irt_x86_64.nexe > /dev/null \
+		|| die "removing execstack flag failed"
+
+	newicon opt/Gitter/linux"${arch}"/logo.png gitter.png
+	newicon -s 256 opt/Gitter/linux"${arch}"/logo.png gitter.png
+	domenu opt/Gitter/linux"${arch}"/gitter.desktop
+
+	insinto /opt/gitter
+	doins -r opt/Gitter/linux"${arch}"/.
+	fperms +x /opt/gitter/Gitter
+	dosym ../../opt/gitter/Gitter /usr/bin/gitter
+
+	pax-mark -m "${ED}"/opt/gitter/Gitter
+}
+
+pkg_preinst() {
+	xdg_icon_cache_update
+}
```

Comment:
Did you test your ebuild after making these changes? Because...
```
>>> Installing (1 of 1) net-im/gitter-bin-4.1.0-r1::testworld
 * checking 145 files for package collisions
>>> Merging net-im/gitter-bin-4.1.0-r1 to /
 * ERROR: net-im/gitter-bin-4.1.0-r1::testworld failed (preinst phase):
 *   xdg_icon_cache_update must be used in pkg_post* phases.
 *
 * Call stack:
 *     ebuild.sh, line  125:  Called pkg_preinst
 *   environment, line 2121:  Called xdg_icon_cache_update
 *   environment, line 3140:  Called die
 * The specific snippet of code:
 *           die "xdg_icon_cache_update must be used in pkg_post* phases.";
 *
 * If you need support, post the output of `emerge --info '=net-im/gitter-bin-4.1.0-r1::testworld'`,
 * the complete build log and the output of `emerge -pqv '=net-im/gitter-bin-4.1.0-r1::testworld'`.
 * The complete build log is located at '/tmp/portage/net-im/gitter-bin-4.1.0-r1/temp/build.log'.
 * The ebuild environment file is located at '/tmp/portage/net-im/gitter-bin-4.1.0-r1/temp/environment'.
 * Working directory: '/tmp/portage/net-im/gitter-bin-4.1.0-r1/homedir'
 * S: '/tmp/portage/net-im/gitter-bin-4.1.0-r1/work'
!!! FAILED preinst: 1

>>> Failed to install net-im/gitter-bin-4.1.0-r1
```
There are also multiple QA issues, 
```
 * QA Notice: The following files contain writable and executable sections
 *  Files with such sections will not work properly (or at all!) on some
 *  architectures/operating systems.  A bug should be filed at
 *  https://bugs.gentoo.org/ to make sure the issue is fixed.
 *  For more information, see:
 *
 *    https://wiki.gentoo.org/wiki/Hardened/GNU_stack_quickstart
 *
 *  Please include the following list of files in your report:
 *  Note: Bugs should be filed for the respective maintainers
 *  of the package in question and not hardened@gentoo.org.
 * !WX --- --- opt/gitter/pnacl/pnacl_public_x86_64_libcrt_platform_a:lib_a-memcpy.o
 * !WX --- --- opt/gitter/pnacl/pnacl_public_x86_64_libcrt_platform_a:lib_a-memset.o
 * !WX --- --- opt/gitter/pnacl/pnacl_public_x86_64_libcrt_platform_a:setjmp.o


 * QA Notice: Files built without respecting CFLAGS have been detected
 *  Please include the following list of files in your report:
 * /opt/gitter/minidump_stackwalk
 * /opt/gitter/nacl_helper_bootstrap


 * QA Notice: Files built without respecting LDFLAGS have been detected
 *  Please include the following list of files in your report:
 * /opt/gitter/minidump_stackwalk
```

the later ones are an easy fix with QA_PREBUILT. You can replicate these with https://wiki.gentoo.org/wiki/Package_testing#make.conf

Reply:
pkg_preinst removed.
As for QA I can't remove them with fix-gnustack

Reply:
You can still fix
```
 * QA Notice: Files built without respecting CFLAGS have been detected
 *  Please include the following list of files in your report:
 * /opt/gitter/minidump_stackwalk
 * /opt/gitter/nacl_helper_bootstrap
 * QA Notice: Files built without respecting LDFLAGS have been detected
 *  Please include the following list of files in your report:
 * /opt/gitter/minidump_stackwalk
```

easily.

Reply:
I don't see this QA and have never seen them before.

Part of my build.log:

```
 * Final size of build directory: 259384 KiB (253.3 MiB)
 * Final size of installed tree:  259392 KiB (253.3 MiB)


 * QA Notice: The following files contain writable and executable sections
 *  Files with such sections will not work properly (or at all!) on some
 *  architectures/operating systems.  A bug should be filed at
 *  https://bugs.gentoo.org/ to make sure the issue is fixed.
 *  For more information, see:
 * 
 *    https://wiki.gentoo.org/wiki/Hardened/GNU_stack_quickstart
 * 
 *  Please include the following list of files in your report:
 *  Note: Bugs should be filed for the respective maintainers
 *  of the package in question and not hardened@gentoo.org.
 * !WX --- --- opt/gitter/pnacl/pnacl_public_x86_64_libcrt_platform_a:lib_a-memcpy.o
 * !WX --- --- opt/gitter/pnacl/pnacl_public_x86_64_libcrt_platform_a:lib_a-memset.o
 * !WX --- --- opt/gitter/pnacl/pnacl_public_x86_64_libcrt_platform_a:setjmp.o

making executable: /opt/gitter/lib/libffmpeg.so
making executable: /opt/gitter/lib/libnode.so
making executable: /opt/gitter/lib/libnw.so
strip: x86_64-pc-linux-gnu-strip --strip-unneeded -N __gentoo_check_ldflags__ -R .comment -R .GCC.command.line -R .note.gnu.gold-version
   /opt/gitter/pnacl/pnacl_public_x86_64_crtbegin_o
   /opt/gitter/nacl_helper_bootstrap
   /opt/gitter/pnacl/pnacl_public_x86_64_crtend_o
   /opt/gitter/minidump_stackwalk
   /opt/gitter/pnacl/pnacl_public_x86_64_crtbegin_for_eh_o

>>> Installing (1 of 1) net-im/gitter-bin-4.1.0-r2::local

```

How can I see this QA ?

Reply:
Use these settings to test your package, 
https://wiki.gentoo.org/wiki/Package_testing#make.conf

Reply:
Thanks. Fixed.